packages/frontend-app-api/src/wiring/parameters.ts

```diff
@@ -196,18 +196,30 @@ export function mergeExtensionParameters(options: {
 }): ExtensionInstanceParameters[] {
   const { sources, builtinExtensions, parameters } = options;
 
+  const pluginExtensions = sources.flatMap(source =>
+    source.extensions.map(extension => ({ ...extension, source })),
+  );
+
+  // Prevent root override
+  if (pluginExtensions.some(({ id }) => id === 'root')) {
+    const rootPluginIds = pluginExtensions
+      .filter(({ id }) => id === 'root')
+      .map(({ source }) => source.id);
+    throw new Error(
+      `The following plugins are overriding root extensions and root extensions cannot be overridden: ${rootPluginIds}`,
```

Comment:
```suggestion
      `The following plugin(s) are overriding the 'root' extension which is forbidden: ${rootPluginIds.join(',')}`,
```
Maybe :shrug:

I believe root is always a single extension

Reply:
Done 🙌🏻 : https://github.com/backstage/backstage/pull/20032/commits/88226b9e824574173823bba2ee56c5eafd1f86f8